### PR TITLE
Ginger linux transformation

### PR DESCRIPTION
**Pull request by uzanh**
Act validation & Email migration

### CODE REVIEW

File: Ginger/GingerCore/GingerCore.csproj [hunk 1]
```diff
@@ -638,6 +636,7 @@
     <None Include="Resources\Pin32x32.png" />
   </ItemGroup>
   <ItemGroup>
+    <Folder Include="Actions\Communication\" />
```
Comment (MeniKadosh1) [inline Ginger/GingerCore/GingerCore.csproj:639]: do we need this folder?

File: Ginger/GingerCoreNET/ActionsLib/ActValidation.cs [hunk 2]
```diff
@@ -16,22 +16,40 @@
...
+        {
+            VBS,
+            CS
+        }
+        
+        public eCalcEngineType CalcEngineType
+        {
+            get
+            {
+                //return (eCalcEngineType)GetInputParamValue<eCalcEngineType>(nameof(CalcEngineType));
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/ActionsLib/ActValidation.cs:44]: remove commented out code

File: Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActeMailTests.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActeMailTests.cs:1]: it's empty so no need to add for now

File: Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActValidationTests.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            ActValidation actValidation = new ActValidation();
+            actValidation.ValueExpression = new ValueExpression(actValidation, "");
+
+            //Act
+            actValidation.CalcEngineType = ActValidation.eCalcEngineType.CS;
+            actValidation.Condition = "4=4";
+            actValidation.Execute();
+
+            //Assert
+            Assert.AreEqual(actValidation.Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Failed);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActValidationTests.cs:63]: validate also act.ExInto to validate the Failed reason

File: Ginger/GingerCoreNET/GeneralLib/Email.cs [hunk 5]
```diff
@@ -410,6 +410,11 @@ public bool Send_SMTP()
             }
         }
 
+        private string ReplaceToCrossOSSeparator(string AttachmentFileName)
+        {
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/GeneralLib/Email.cs:414]: Do it only if not running on Windows

File: Ginger/GingerCore/GingerCore.csproj (hunk 1 above)
Comment (uzanh) [reply Ginger/GingerCore/GingerCore.csproj:639]: removed

File: Ginger/GingerCoreNET/ActionsLib/ActValidation.cs (hunk 2 above)
Comment (uzanh) [reply Ginger/GingerCoreNET/ActionsLib/ActValidation.cs:44]: done

File: Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActeMailTests.cs (hunk 3 above)
Comment (uzanh) [reply Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActeMailTests.cs:1]: done

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**